### DISCUSSION TITLE
generate conformers with protonation states

### DISCUSSION THREAD

**Discussion by YoavShamir5** [rdkit/rdkit#7048] [Q&A]
Hello, 

I am interested in generating conformers for different protonation states of the same compound. In the following example, it seems like enumerating protomer SMILES and then using AddHs is the way to go prior to embeding and getting the conformers. I would love to know if this would be the best way to generate conformers of specific protomers via RDKit:

from dimorphite_dl import DimorphiteDL
from rdkit import Chem
from rdkit.Chem import Draw

dimorphite_dl = DimorphiteDL(

    min_ph=7.4,
    max_ph=7.4,
    max_variants=128,
    label_states=False,
    pka_precision=1.0
)

prot_lst = dimorphite_dl.protonate('C=CC(=O)N1CCC[C@H](C1)N2C3=C(C(=N2)C4=CC=C(C=C4)OC5=CC=CC=C5)C(=NC=N3)N')
print(prot_lst)

**['C=CC(=O)N1CCC[C@@H](n2nc(-c3ccc(Oc4ccccc4)cc3)c3c(N)ncnc32)C1', 'C=CC(=O)[NH+]1CCC[C@@H](n2nc(-c3ccc(Oc4ccccc4)cc3)c3c(N)ncnc32)C1']**

mol_lst = []

for smiles in prot_lst:
    mol = Chem.MolFromSmiles(smiles)
    mol_lst.append(mol)
    charge = Chem.rdmolops.GetFormalCharge(mol)
    print("charge: ", charge)

Draw.MolsToGridImage(mol_lst , molsPerRow=2, subImgSize=(500,500))
![image](https://github.com/rdkit/rdkit/assets/39048833/54448f1a-c83e-452b-a631-3e31b3cf5d37)

new_mol_lst = []
for mol in mol_lst:
    mol_new = Chem.AddHs(mol)
    new_mol_lst.append(mol_new)

Draw.MolsToGridImage(new_mol_lst , molsPerRow=2, subImgSize=(500,500))

![image](https://github.com/rdkit/rdkit/assets/39048833/f928150a-27d0-4d68-a78c-fbc3f60ec038)

**Comment by bp-kelley** [comment]
This is the way I would do it with the RDKit.

**Comment by bp-kelley** [comment]
One issue you may face is introducing stereo centers.  This can be problematic and might need to be resolved.

**Reply by YoavShamir5** [reply]
Thanks for the input! Could you please elaborate what do you think could be problematic in this method of generating protonated conformers when the compound has stereocenters?
